src/OSPSuite.Presentation/Presenters/Charts/ChartDisplayPresenter.cs

```diff
@@ -557,20 +570,31 @@ public void ShowCurveInLegend(Curve curve, bool show)
       public void MoveSeriesInLegend(Curve movingCurve, Curve targetCurve)
       {
          Chart.MoveCurvesInLegend(movingCurve, targetCurve);
-         updateChart();
+         View.ReOrderLegend();
       }
 
       public void SetNoCurvesSelectedHint(string hint)
       {
          _view.SetNoCurvesSelectedHint(hint);
       }
 
-      public void Handle(ChartUpdatedEvent chartUpdatedEvent)
+      public void Handle(ChartUpdatedEvent chartUpdatedEvent) => handleChartUpdate(chartUpdatedEvent, Chart.Curves, curveDataChanged: true);
+
+      public void Handle(CurveChartUpdatedEvent chartUpdatedEvent) => handleChartUpdate(chartUpdatedEvent, chartUpdatedEvent.CurvesToUpdate, chartUpdatedEvent.CurveDataChanged);
+
+      private void handleChartUpdate(ChartUpdatedEvent chartUpdatedEvent, IReadOnlyCollection<Curve> curvesToUpdate, bool curveDataChanged)
       {
          if (!canHandle(chartUpdatedEvent))
             return;
 
-         updateChart();
+         if (Chart.BatchUpdateEnabled)
+         {
+            _curveBinders.Each(x => x.HideAllSeries());
+            View.ShowWatermark(Captions.RefreshChartToUpdate);
+            return;
```

Comment:
In the batch update mode, when the first change is handled, the series are all hidden and the watermark drawn. That happens very fast. Any time I try to preserve the outdated chart with a  watermark, there's a significant price to pay. This is probably just the paint time when adding the watermark itself. The chart content is repainted.

I'm jus not sure it's acceptable that the chart is blank when you make batch changes. You don't have any outdated context... Maybe it's ok.

Reply:
yeah, the diagram is gone in this version when the batch mode is on, See the screenshots.